### PR TITLE
support `LargeList` for `arrow_cast`, support `ScalarValue::LargeList`

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Closes #.

## Rationale for this change
It is hard to test `LargeList` in sqllogictest #8121, the convenient idea is to use `arrow_cast()` instead of `make_array`

## What changes are included in this PR?
- support `LargeList` for `arrow_cast`
- support `ScalarValue::LargeList`
- support `ScalarValue::new_large_list`
- support `LargeList` for protobuf

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @Weijun-H  -- this looks very nice 👍

File: datafusion/common/src/scalar.rs [hunk 1]
```diff
@@ -342,7 +346,38 @@ impl PartialOrd for ScalarValue {
...
                 }
             }
+            (LargeList(arr1), LargeList(arr2)) => {
+                if arr1.data_type() == arr2.data_type() {
+                    let list_arr1 = as_large_list_array(arr1);
+                    let list_arr2 = as_large_list_array(arr2);
+                    if list_arr1.len() != list_arr2.len() {
+                        return None;
+                    }
+                    for i in 0..list_arr1.len() {
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/common/src/scalar.rs:357]: Maybe you can clean it up in a follow on PR (by asserting the list is length 1)

**Comment by alamb** [comment]
Thanks again @Weijun-H

File: datafusion/common/src/scalar.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Weijun-H) [reply datafusion/common/src/scalar.rs:357]: I will work on this on next PR

[the remark this answers is not in the document]
Comment (Weijun-H) [reply datafusion/common/src/scalar.rs:357]: I think @jayzhan211 is working on this #8253

**Pull request merged**